lib/bns/fetcher/postgres/types/response.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module Fetcher
+  module Postgres
+    module Types
+      ##
+      # Represents a response received from the Postgres API. It encapsulates essential information about the response,
+      # providing a structured way to handle and analyze it's responses.
+      class Response
+        attr_reader :status, :message, :response, :fields, :records
+
+        SUCCESS_STATUS = "PGRES_TUPLES_OK"
+
+        def initialize(response)
+          if response.res_status == SUCCESS_STATUS
+            success_response(response)
+          else
+            unsuccess_response(response)
+          end
+        end
+
+        private
+
+        def success_response(response)
+          @status = response.res_status
+          @message = "success"
+          @response = response
+          @fields = response.fields
+          @records = response.values
+        end
+
+        def unsuccess_response(response)
```

Comment:
I think it should be `unsuccessful_response`, however, maybe we can find a better name for this method? something like `error_response` or `failure_response`?